packages/earn-controller/src/EarnController.ts

```diff
@@ -0,0 +1,334 @@
+import { Web3Provider } from '@ethersproject/providers';
+import type {
+  AccountsControllerGetSelectedAccountAction,
+  AccountsControllerSelectedAccountChangeEvent,
+} from '@metamask/accounts-controller';
+import type {
+  ControllerGetStateAction,
+  ControllerStateChangeEvent,
+  RestrictedControllerMessenger,
+  StateMetadata,
+} from '@metamask/base-controller';
+import { BaseController } from '@metamask/base-controller';
+import { convertHexToDecimal } from '@metamask/controller-utils';
+import type { NetworkControllerNetworkDidChangeEvent } from '@metamask/network-controller';
+import type {
+  NetworkControllerGetNetworkClientByIdAction,
+  NetworkControllerGetStateAction,
+} from '@metamask/network-controller';
+import {
+  StakeSdk,
+  StakingApiService,
+  type PooledStake,
+  type StakeSdkConfig,
+  type VaultData,
+} from '@metamask/stake-sdk';
+
+export const controllerName = 'EarnController';
+
+// === Enums and Identifiers ===
+export enum EarnProductType {
+  POOLED_STAKING = 'pooled_staking',
+  STABLECOIN_LENDING = 'stablecoin_lending',
+}
+
+// === Product Types ===
+export type PooledStakingProduct = {
+  pooledStakes: PooledStake;
+  exchangeRate: string;
+  vaultData: VaultData;
+  isEligible: boolean;
+};
+
+export type StablecoinLendingProduct = {
+  vaults: StablecoinVault[];
+};
+
+export type StablecoinVault = {
+  symbol: string;
+  name: string;
+  chainId: number;
+  tokenAddress: string;
+  vaultAddress: string;
+  currentAPY: string;
+  supply: string;
+  liquidity: string;
+};
+
+/**
+ * Metadata for the EarnController.
+ */
+export const earnControllerMetadata: StateMetadata<EarnControllerState> = {
+  [EarnProductType.POOLED_STAKING]: {
```

Comment:
You might not want to tie the metadata and state properties to enums or other variables. If those variables change then you would be changing your state properties which would result in a breaking change without you knowing.